database/src/episode.rs

```diff
@@ -515,126 +212,38 @@ impl InsertableEpisode {
     /// * `conn` - diesel connection reference to postgres
     /// * `show_id` - the id of the tv show we are trying to asociate this episode with
     ///
-    /// # Example
-    /// ```
-    /// use database::get_conn_devel as get_conn;
-    /// use database::library::{Library, InsertableLibrary, MediaType};
-    /// use database::media::{InsertableMedia, Media};
-    /// use database::tv::InsertableTVShow;
-    /// use database::season::InsertableSeason;
-    /// use database::episode::{InsertableEpisode, Episode};
-    ///
-    /// let conn = get_conn().unwrap();
-    ///
-    /// let library = InsertableLibrary {
-    ///     name: "test".into(),
-    ///     location: "/dev/null".to_string(),
-    ///     media_type: MediaType::Tv,
-    /// };
-    ///
-    /// let library_id = library.insert(&conn).unwrap();
-    ///
-    /// let new_show = InsertableMedia {
-    ///     library_id: library_id,
-    ///     name: "test".into(),
-    ///     added: "test".into(),
-    ///     media_type: MediaType::Tv,
-    ///     ..Default::default()
-    /// };
-    ///
-    /// let show_id = new_show.into_static::<InsertableTVShow>(&conn).unwrap();
-    ///
-    /// let new_season = InsertableSeason {
-    ///     season_number: 1,
-    ///     added: "test".into(),
-    ///     poster: "test".into(),
-    /// };
-    ///
-    /// let season_id = new_season.insert(&conn, show_id).unwrap();
-    ///
-    /// let new_episode = InsertableEpisode {
-    ///     media: InsertableMedia {
-    ///         library_id: library_id,
-    ///         name: "test_episode".into(),
-    ///         added: "test".into(),
-    ///         media_type: MediaType::Episode,
-    ///         ..Default::default()
-    ///     },
-    ///     seasonid: season_id,
-    ///     episode: 1,
-    /// };
-    ///
-    /// let show = Media::get(&conn, show_id).unwrap();
-    /// let all_episodes = Episode::get_all_of_tv(&conn, &show).unwrap();
-    /// assert!(all_episodes.len() == 0);
-    ///
-    /// let episode_id = new_episode.insert(&conn, show_id).unwrap();
-    ///
-    /// let show = Media::get(&conn, show_id).unwrap();
-    /// let all_episodes = Episode::get_all_of_tv(&conn, &show).unwrap();
-    /// assert!(all_episodes.len() == 1);
-    ///
-    /// let episode = &all_episodes[0];
-    /// assert_eq!(episode.id, episode_id);
-    /// assert_eq!(episode.episode, 1);
-    ///
-    /// Library::delete(&conn, library_id).unwrap();
     pub async fn insert(
         &self,
         conn: &crate::DbConnection,
-        id: i32,
-        media_id: i32,
-    ) -> Result<i32, DatabaseError> {
-        use crate::schema::season;
-        use crate::schema::tv_show;
-
-        let _tv_show = tv_show::dsl::tv_show
-            .find(id)
-            .get_result_async::<TVShow>(conn)
+        tv_id: i64,
+        media_id: i64,
+    ) -> Result<i64, DatabaseError> {
```

Comment:
all of the queries in here should be executed from inside a transaction, otherwise theres a very high chance the scanners will crash or insert invalid data into the database because of phantom reads.